app/serializers/alchemy/page_tree_serializer.rb

```diff
@@ -46,27 +46,46 @@ def pages
     protected
 
     def page_hash(page, has_children, level, folded)
-      {
+      p_hash = {
         id: page.id,
         name: page.name,
-        permissions: page_permissions(page, opts[:ability]),
         public: page.public?,
         visible: page.visible?,
         restricted: page.restricted?,
-        status_titles: page_status_titles(page),
         page_layout: page.page_layout,
         slug: page.slug,
         redirects_to_external: page.redirects_to_external?,
-        locked: page.locked?,
-        definition_missing: page.definition.blank?,
         urlname: page.urlname,
-        external_urlname: page.external_urlname,
+        external_urlname: page.redirects_to_external? ? page.external_urlname : nil,
```

Comment:
Why is that if-else needed? Couldn't we leave it as it is because  `page.external_urlname` would be `nil` anyways?